seat.c

```diff
@@ -684,6 +664,10 @@ handle_start_drag(struct wl_listener *listener, void *data)
 	}
 	drag_icon->seat = seat;
 	drag_icon->wlr_drag_icon = wlr_drag_icon;
+	drag_icon->scene_node = wlr_scene_subsurface_tree_create(&seat->server->scene->node, wlr_drag_icon->surface);
+	if (!drag_icon->scene_node) {
+		return;
```

Comment:
Doesn't returning early here means we will never clean up this drag icon? It hasn't been added to the `seat->drag_icons` list yet, nor has its `destroy` handler been setup.

Reply:
Added logic to free the drag icon.